common/app/model/Badges.scala

```diff
@@ -3,10 +3,40 @@ package model
 import conf.switches.Switches
 import conf.{Configuration, Static}
 import layout.FaciaContainer
+import java.security.MessageDigest
+import java.math.BigInteger
 
-case class Badge(seriesTag: String, imageUrl: String, classModifier: Option[String] = None)
+
+trait BaseBadge{
+  def imageUrl(id: String): String
+  def classModifier: Option[String]
+  def isBadgeTag(tag: String):Boolean
+}
+
+case class Badge(seriesTag: String, staticImageUrl: String, classModifier: Option[String] = None) extends BaseBadge {
+  def isBadgeTag(tag: String) = seriesTag == tag
+  def imageUrl(id: String) = staticImageUrl
+}
+case class SpecialBadge(hashedTag: String, classModifier: Option[String] = None) extends BaseBadge {
+  def isBadgeTag(tag: String) = md5(salt + tag).contains(hashedTag)
+  def imageUrl(id: String) = s"https://assets.guim.co.uk/special/$id/special-badge.svg"
```

Comment:
by the time we call imageUrl we have the Badge instance and we're happy it is the right one